autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/importpassword/ImportPasswordsViewModel.kt

```diff
@@ -38,6 +56,54 @@ class ImportPasswordsViewModel @Inject constructor(
         }
     }
 
+    /**
+     * When user launches this screen we want to record the time they started the user-journey.
+     *
+     * Store timestamp and start the user-journey success clock if:
+     *     sync has NOT been enabled
+     *     or sync is enabled but not synced with any desktop devices
+     */
+    fun userLaunchedScreen() {
+        Timber.v("User launched import passwords screen; checking user-journey status")
+
+        // use appCoroutineScope as the call to get account state might not be quick; we want the call to succeed even if the user leaves the screen
+        appCoroutineScope.launch(dispatchers.io()) {
+            if (aUserJourneyIsOngoing()) {
```

Comment:
Wouldn't be better to move all logic related to user journey checks into it's own class?

I see few benefits:
- ViewModel will be simplified of all logic about sync journey checks, which can be extracted. We can move all that business logic and just keep UI logic here.
- Dependencies can also be simplified: ViewModel has journey detecting logic which relies on `ImportPasswordsViaDesktopSyncDataStore` and `UserJourneyEndRecorder` and `DeviceSyncState`. `UserJourneyEndRecoder` on a different class (with dependency to `ImportPasswordsViaDesktopSyncDataStore`).

I think the flow is easier if we just do: viewModel -> SyncImportUserJourney -> Business logic related to tracking journey

Reply:
Yeah, I think that is a good suggestion. I am going to do that as a follow-up, tracking as [Autofill: extract more of the business logic around importing passwords via desktop to its own class](https://app.asana.com/0/1207721513161801/1207721513161801/f)